src/Features/Blockcore.Features.Miner/Staking/PosMinting.cs

```diff
@@ -718,8 +718,9 @@ public async Task<bool> CreateCoinstakeAsync(List<UtxoStakeDescription> utxoStak
             }
 
             // Get reward for newly created block.
-            long reward = fees + this.consensusManager.ConsensusRules.GetRule<CheckPosUtxosetRule>().GetProofOfStakeReward(chainTip.Height + 1);
-            if (reward <= 0)
+            var checkPosUtxosetRule = this.consensusManager.ConsensusRules.GetRule<CheckPosUtxosetRule>();
+            long reward = fees + checkPosUtxosetRule.GetProofOfStakeReward(chainTip.Height + 1);
+            if (reward <= 0 && checkPosUtxosetRule.RewardRequired())
```

Comment:
This is interesting, if reward is not required just set it to zero in Network settings why add a flag?

Reply:
Default is set to `reward <= 0` and with x42 a block might have no rewards, and this fails, so I need the flag here.

Reply:
I can add another network consensus item similar to the blockmintxfee, is that what you are thinking here?

Reply:
If we need this then yes a network consensus is better then changing the rule.
However I am wondering if it should not be `reward < 0`

Reply:
Agreed.

Reply:
yes I think we should change this condition to be `reward < 0` it's left over code form stratisx I believe and it does not make sense anymore

Reply:
Done.